sql/analyzer/apply_indexes_for_subquery_comparisons.go

```diff
@@ -31,52 +72,110 @@ import (
 // 4. The Child is a *plan.ResolvedTable.
 // 5. The referenced field in the Child is indexed.
 func applyIndexesForSubqueryComparisons(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
-	aliases, err := getTableAliases(n, scope)
-	if err != nil {
-		return nil, transform.SameTree, err
-	}
+	var recurse func(n sql.Node, inScope *aliasScope) (sql.Node, transform.TreeIdentity, error)
+	recurse = func(n sql.Node, inScope *aliasScope) (sql.Node, transform.TreeIdentity, error) {
+		var alias, target string
+		var err error
+		switch n := n.(type) {
+		case *plan.TableAlias:
+			alias = n.Name()
+			switch c := n.Child.(type) {
+			case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode, *plan.RecursiveTable, *plan.DeferredAsOfTable:
```

Comment:
Is it worth using the `sql.Table` interface to capture table types more aggressively? For example... would other implementations like `FilteredTable` be valid here? (not sure if we'd push a filter down through a TableAlias currently or not, just brainstorming)

Reply:
i think `sql.Table` is normally under one of these nodes, not a plan node itself, although the main two `ResolvedTable` and `IndexedTableAccess` do implement `sql.Table`.Mmight need a different interface i the future to capture TableNode